misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/IterableAssertionsSpec.kt

```diff
@@ -126,4 +131,35 @@ abstract class IterableAssertionsSpec(
         }
     }
 
+
+    describeFun(containsNoDuplicates) {
+        val containsNoDuplicatesFun = containsNoDuplicates.lambda
+
+        it("list without duplicates") {
+            expect(listOf(1, 2) as Iterable<Int>).containsNoDuplicatesFun()
+        }
+
+        it("list with duplicates") {
+            val input = listOf(1, 2, 1, 2, 3, 4, 4, 4)
+            val duplicates = input
+                .mapWithIndex()
+                .filter { (_, element) ->
+                    input.count { e -> e == element } > 1
+                }
+                .map { (index, element) -> index.toString() to element.toString() }
+
+            val expectedErrors = duplicates
+                .map { (index, element) -> indexElements.format("$index: $element") }
+
+            expect {
+                expect(input as Iterable<Int>).containsNoDuplicatesFun()
+            }.toThrow<AssertionError> { messageContains("$hasDescriptionBasic: $duplicateElements") }
+
+            expectedErrors.forEach { expectedMessage ->
+                expect {
+                    expect(input as Iterable<Int>).containsNoDuplicatesFun()
+                }.toThrow<AssertionError> { messageContains(expectedMessage) }
+            }
```

Comment:
I prefer to not repeat logic in tests but keep it simple and stupid. 
Moreover, you can assert everything in one expect:
```suggestion
            fun index(i: Int, element: Int) = DescriptionIterableAssertion.INDEX.getDefault().format("$i: $element"): 
            
            val input = listOf(1, 2, 1, 2, 3, 4, 4, 4).asIterable()
            expect {
                expect(input).containsNoDuplicatesFun()
            }.toThrow<AssertionError> { 
              message {
                contains("$hasDescriptionBasic: $duplicateElements") 
                contains(index(0, 1), index(1, 2), index(2, 1), index(3, 2), index(5, 4), index(6, 4), index(7, 4))
                containsNot(TODO("what should not be in here?"))
              }
            }
```
IMO way cleaner

Reply:
ok np, will do that